src/test/java/tools/jackson/databind/ser/CanonicalBigDecimalSerializer.java

```diff
@@ -0,0 +1,61 @@
+package tools.jackson.databind.ser;
+
+import java.math.BigDecimal;
+
+import tools.jackson.core.JacksonException;
+import tools.jackson.core.JsonGenerator;
+import tools.jackson.databind.SerializerProvider;
+import tools.jackson.databind.ser.std.StdSerializer;
+
+public class CanonicalBigDecimalSerializer extends StdSerializer<BigDecimal>
+        implements ValueToString<BigDecimal> {
+
+    protected CanonicalBigDecimalSerializer() {
+        super(BigDecimal.class);
+    }
+
+    @Override
+    public void serialize(BigDecimal value, JsonGenerator gen, SerializerProvider provider)
+            throws JacksonException {
+        CanonicalNumberGenerator.verifyBigDecimalRange(value, provider);
+
+        String output = convert(value);
+        gen.writeNumber(output);
+    }
+
+    @Override
+    public String convert(BigDecimal value) {
+        // TODO Convert to exponential form if necessary
+        BigDecimal stripped = value.stripTrailingZeros();
+        int scale = stripped.scale();
+        String text = stripped.toPlainString();
+        if (scale == 0) {
+            return text;
+        }
+
+        int pos = text.indexOf('.');
+        int exp;
+        if (pos >= 0) {
+            exp = pos - 1;
+
+            if (exp == 0) {
+                return text;
+            }
+
+            text = text.substring(0, pos) + text.substring(pos + 1);
+        } else {
+            exp = -scale;
+            int end = text.length();
+            while (end > 0 && text.charAt(end - 1) == '0') {
+                end --;
+            }
+            text = text.substring(0, end);
+        }
+
+        if (text.length() == 1) {
+            return text + 'E' + exp;
+        }
+
+        return text.substring(0, 1) + '.' + text.substring(1) + 'E' + exp;
+    }
```

Comment:
I created this from scratch since I couldn't find anything suitable on the classpath.

The toString() methods in BigDecimal itself don't work (they either don't print an exponent or print only some numbers with exponent) and `DecimalFormat` has weird quirks (like being not thread safe which might or might not cause problems). There are other implementations of `DecimalFormat` in some Apache commons library but I didn't want to add new dependencies.

This class only exists because https://gibson042.github.io/canonicaljson-spec/ requires it. Personally, I would always print the number in a readable form without exponent OR as an integer with a negative exponent (so it can always be parsed without any rounding errors).

I would be grateful for advice how to proceed here.